### ISSUE TITLE
[UX] Make "This action cannot be undone!" in confirmation forms a warning

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3422]
Before:

<img width="611" alt="screen shot 2018-12-12 at 5 43 13 am" src="https://user-images.githubusercontent.com/2423362/49845795-dd24c700-fdd0-11e8-9732-de31b1a0cc52.png">

**Comment by klonos** [comment]
...after:

<img width="700" alt="screen shot 2018-12-12 at 5 50 29 am" src="https://user-images.githubusercontent.com/2423362/49846018-e4000980-fdd1-11e8-92c0-c20bbe1fda18.png">

Note that you will not be able to actually test this in the PR sandbox, as the caption/description in `node_type_delete_confirm()` is already overridden (with the default text "This action cannot be undone.", which is really silly btw): https://github.com/backdrop/backdrop/blob/1.x/core/modules/node/node.types.inc#L644

**Comment by klonos** [comment]
...this is a small, harmless change (as in that it only changes the default message provided by [`confirm_form()`](https://api.backdropcms.org/api/backdrop/core%21modules%21system%21system.module/function/confirm_form/1) **only if no `$description` argument is specified**). Nevertheless, I did a quick search through our codebase and I found it really annoying/silly that we are repeatedly overriding `$description` with `t('This action cannot be undone.')`, which is the default value of `$description` anyway 🙄

So note to self: either create an alternative PR here to replace `confirm_form()` with a new function that works better (and then make `confirm_form()` a wrapper function for that new function for backwards compatibility), or if this gets merged as is, then file a follow-up issue and do it there. DX++

**Comment by docwilmot** [comment]
This is a reasonable change, but if "we are repeatedly overriding $description", we should go through and replace all those with NULL so that the default will be used, for consistency sake.

**Comment by jenlampton** [comment]
I agree, I like this change but think we should be doing it correctly as you both recommend. Marking as NW. I'm also removing the milestone, as I'm not sure how/when it was added, but I will add the tag for milestone review.

**Comment by klonos** [comment]
Thanks for the review/feedback @docwilmot and @jenlampton ...sorry for taking so long to come back to this. Let's get this done before reworking #2461
